processor/lookupprocessor/processor.go

```diff
@@ -0,0 +1,301 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lookupprocessor
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+// lookupProcessor is a lookupProcessor that looks up values and adds them to telemetry
+type lookupProcessor struct {
+	logger  *zap.Logger
+	csvFile *CSVFile
+	context string
+	field   string
+	cancel  context.CancelFunc
+}
+
+// newLookupProcessor creates a new lookupProcessor
+func newLookupProcessor(cfg *Config, logger *zap.Logger) *lookupProcessor {
+	return &lookupProcessor{
+		logger:  logger,
+		csvFile: NewCSVFile(cfg.CSV, cfg.Field),
+		context: cfg.Context,
+		field:   cfg.Field,
+	}
+}
+
+// start starts the processor
+func (p *lookupProcessor) start(_ context.Context, _ component.Host) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	p.cancel = cancel
+	go p.loadCSV(ctx)
```

Comment:
We should put wait groups around this. Since the collector is an in memory process if this goroutine takes a long time to shut down or doesn't at all it can leak.